packages/@aws-cdk/aws-appsync/lib/graphqlapi.ts

```diff
@@ -292,6 +305,16 @@ export interface GraphqlApiProps {
    * @default - false
    */
   readonly xrayEnabled?: boolean;
+
+  /**
+   * The domain name configuration for the GraphQL API
+   *
+   * The hosted zone and CName must be configured in addition to this setting to
+   * enable custom domain URL
+   *
+   * @default - none a unique name is generated
+   */
+  readonly domainName?: DomainOptions;
```

Comment:
Wondering about this name... thoughts on domainOptions ?

Reply:
I actually feel like the `DomainOptions` property is misnamed -- both properties under `DomainOptions` are required, and we usually use 'options' to signify optional properties. Instead, I like the property name `domainName`, and I think the type should be `DomainName` as well.

Reply:
using `DomainName` would cause the following JSII warning

```
lib/graphqlapi.ts:270:12 - warning JSII5019: The property name "domainName" conflicts with the declaring interface "DomainName". This will result in renaming the interface to "
_DomainName" in C#. Consider renaming "domainName".
```

`CustomDomainOptions` is used in [aws-cognito](https://github.com/aws/aws-cdk/blob/7966f8d48c4bff26beb22856d289f9d0c7e7081d/packages/%40aws-cdk/aws-cognito/lib/user-pool-domain.ts#L59).  So not unprecedented.

Reply:
I think a closer comparison is rest api which Used `DomainNameOptions` as seen [here](https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-apigateway/lib/restapi.ts#L145-L150) I'm good with either solution, but I think it's odd to change the convention between rest and graphql for domain name configuration.